plugins/database/hana/hana_test.go

```diff
@@ -49,38 +55,145 @@ func TestHANA_CreateUser(t *testing.T) {
 		"connection_url": connURL,
 	}
 
+	initReq := newdbplugin.InitializeRequest{
+		Config:           connectionDetails,
+		VerifyConnection: true,
+	}
+
 	db := new()
-	_, err := db.Init(context.Background(), connectionDetails, true)
+	_, err := db.Initialize(context.Background(), initReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	usernameConfig := dbplugin.UsernameConfig{
-		DisplayName: "test-test",
-		RoleName:    "test-test",
+	password, err := credsutil.RandomAlphaNumeric(32, true)
+	if err != nil {
+		t.Fatalf("failed to generate password: %s", err)
+	}
+	password = strings.Replace(password, "-", "_", -1)
+
+	req := newdbplugin.NewUserRequest{
+		UsernameConfig: newdbplugin.UsernameMetadata{
+			DisplayName: "test-test",
+			RoleName:    "test-test",
+		},
+		Statements: newdbplugin.Statements{
+			Commands: []string{},
+		},
+		Password:   password,
+		Expiration: time.Now().Add(time.Hour),
 	}
 
 	// Test with no configured Creation Statement
-	_, _, err = db.CreateUser(context.Background(), dbplugin.Statements{}, usernameConfig, time.Now().Add(time.Hour))
+	userResp, err := db.NewUser(context.Background(), req)
 	if err == nil {
 		t.Fatal("Expected error when no creation statement is provided")
 	}
 
-	statements := dbplugin.Statements{
-		Creation: []string{testHANARole},
+	// Add a statement command
+	req.Statements.Commands = []string{testHANARole}
+
+	userResp, err = db.NewUser(context.Background(), req)
```

Comment:
Similarly, this can use `dbtesting.AssertNewUser` (and below in the other tests). Note that if you are expecting the error to be not nil then you can't use those assertion functions since they assert the call succeeded.